src/auth.js

```diff
@@ -83,16 +83,18 @@ export function refresh (config) {
   })
 }
 
-export function auth () {
+export function auth (force) {
   return new Promise((resolve, reject) => {
     const configDir = getConfigDir()
     const configPath = join(configDir, 'auth.json')
     let config
 
-    try {
-      config = JSON.parse(readFileSync(configPath, 'ascii'))
-    } catch (e) {
-      // Ignore.
+    if (!force) {
```

Comment:
I would split out a function `fetchAuthFromP4d`. It would allow expressing this more elegantly/explicitly, not relying on config === undefined in the check below. Rather, you could call it right away if force === false

Reply:
Lets keep this for a slightly bigger refactor.